internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/preprocessor/ReactFileDownloader.java

```diff
@@ -0,0 +1,196 @@
+        }
+    }
+
+    /**
+     * Install npm modules which are referenced in package.json files.
+     *
+     * @param path Directory where the package.json file is placed.
+     * @throws DownloaderException exception occurred while downloading NPM Modules.
+     */
+    private static void installNPMModules(String path) throws DownloaderException {
+        Process process = null;
+        try {
+            ProcessBuilder processBuilder = new ProcessBuilder(JSScannerConstants.NPM,
+                    JSScannerConstants.INSTALL);
+            processBuilder.directory(new File(path));
+            process = processBuilder.start();
+            process.waitFor();
+        } catch (IOException | InterruptedException e) {
+            throw new DownloaderException("Error occurred while downloading NPM modules due to : " + e.getMessage());
+        } finally {
+            assert process != null;
+            process.destroyForcibly();
+        }
+    }
+
+    /**
+     * Create Directory
+     *
+     * @param dir Directory to be created
+     */
+    private static void createDirectory(File dir) {
+
+        if (!dir.exists()) {
+            boolean isDirCreated = dir.mkdir();
+            if (!isDirCreated) {
+                log.error((dir.getAbsolutePath() + " is not created"));
+            }
+        }
+    }
+
+    /**
+     * Get component version from product main pom file.
+     *
+     * @param version    product version
+     * @param repoName   reponame of the product
+     * @param versionTag version tag of component
+     * @return component version
+     * @throws ApiInvokerException Exception occurred while API Call.
+     */
+    private static String getComponentVersionFromPom(String version, String repoName, String versionTag) throws
+            ApiInvokerException {
+        String componentVersion = null;
+        //api endpoint to get pom file
+        String url = JSScannerConstants.GIT_SEARCH_URL + JSScannerConstants.POM + "+repo:wso2/" + repoName + "/tree/" +
+                version;
+        String response = CommonApiInvoker.connectGitAPI(url);
+        JSONObject resultJsonObject = new JSONObject(response);
+        //array of pom files in product repo
+        JSONArray itemsArray = resultJsonObject.getJSONArray("items");
+        for (int i = 0, size = itemsArray.length(); i < size; i++) {
+            JSONObject currentItem = itemsArray.getJSONObject(i);
+            if (currentItem.get("path").equals(JSScannerConstants.POM)) {
+                //endpoint to get content of pom file
+                String contentAPIURL = currentItem.getString("url");
+                String contentAPIResponse = CommonApiInvoker.connectGitAPI(contentAPIURL);
+                JSONObject contentResponseObject = new JSONObject(contentAPIResponse);
+                //get encoded content of pom file
+                String encodedContent = contentResponseObject.getString("content");
```

Comment:
Is it guaranteed to have a valid value in 'encodedContent'? What happens if the response is something other than what's anticipated?